File: ejerc.diccionarios/Ejercicio 5/Ejercicio5.py
# Diccionario con asignaturas y sus respectivos créditos
# Cada clave es el nombre de la asignatura y el valor, el número de créditos
asignaturas = {'Matemáticas': 6, 'Física': 4, 'Química': 5}

# Recorrer el diccionario para mostrar cada asignatura con su número de créditos
for asignatura, creditos in asignaturas.items():
    print(f"{asignatura} tiene {creditos} créditos")

# Calcular el total de créditos sumando todos los valores del diccionario
total_creditos = sum(asignaturas.values())

# Mostrar el total de créditos del curso
print(f"El número total de créditos del curso es: {total_creditos}")

File: ejerc.diccionarios/Ejercicio 5/Ejercicio5.java
import java.util.HashMap;
import java.util.Map;

public class CreditosAsignaturas {
    public static void main(String[] args) {
        // Crear un HashMap para almacenar asignaturas con su número de créditos
        Map<String, Integer> asignaturas = new HashMap<>();
        asignaturas.put("Matemáticas", 6); // Matemáticas con 6 créditos
        asignaturas.put("Física", 4);      // Física con 4 créditos
        asignaturas.put("Química", 5);     // Química con 5 créditos

        // Variable para acumular el total de créditos
        int totalCreditos = 0;

        // Recorrer el HashMap con un for-each para mostrar cada asignatura y sumar sus créditos
        for (Map.Entry<String, Integer> entry : asignaturas.entrySet()) {
            System.out.println(entry.getKey() + " tiene " + entry.getValue() + " créditos");
            totalCreditos += entry.getValue(); // Acumula los créditos
        }

        // Mostrar el total de créditos del curso
        System.out.println("El número total de créditos del curso es: " + totalCreditos);
    }
}
